xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -6065,55 +6329,121 @@ fuse_thread_proc(void *data)
            size from 'fuse', which is as of today 128KB. If we bring in
            support for higher block sizes support, then we should be
            changing this one too */
-        iobuf = iobuf_get(this->ctx->iobuf_pool);
+        iobuf_in = iobuf_get(this->ctx->iobuf_pool);
 
         /* Add extra 512 byte to the first iov so that it can
          * accommodate "ordinary" non-write requests. It's not
          * guaranteed to be big enough, as SETXATTR and namespace
          * operations with very long names may grow behind it,
          * but it's good enough in most cases (and we can handle
          * rest via realloc). */
-        iov_in[0].iov_base = GF_MALLOC(
+        iovec_in[0].iov_base = GF_MALLOC(
             sizeof(fuse_async_t) + msg0_size + FUSE_EXTRA_ALLOC,
             gf_fuse_mt_iov_base);
 
-        if (!iobuf || !iov_in[0].iov_base) {
+        if (!iobuf_in || !iovec_in[0].iov_base) {
             gf_log(this->name, GF_LOG_ERROR, "Out of memory");
-            if (iobuf)
-                iobuf_unref(iobuf);
-            GF_FREE(iov_in[0].iov_base);
+            if (iobuf_in)
+                iobuf_unref(iobuf_in);
+            GF_FREE(iovec_in[0].iov_base);
             sleep(10);
             continue;
         }
 
-        iov_in[1].iov_base = iobuf->ptr;
+        iovec_in[1].iov_base = iobuf_in->ptr;
+
+        iovec_in[0].iov_len = msg0_size;
+        iovec_in[1].iov_len = psize;
+
+        pthread_mutex_lock(&priv->sq_mutex);
+        {
+            sqe = io_uring_get_sqe(&priv->ring);
+            if (!sqe) {
+                gf_log("this->name", GF_LOG_ERROR,
+                       "Failed to get the submission queue, %s",
+                       strerror(errno));
+
+                iobuf_unref(iobuf_in);
+                GF_FREE(iovec_in[0].iov_base);
+                sleep(10);
+                continue;
+            }
+            io_uring_prep_readv(sqe, priv->fd, iovec_in, 2, /* offset */ 0);
+
+            sqe->flags |= IOSQE_ASYNC;
+            //            sqe->flags |= IOSQE_FIXED_FILE;
+
+            setctx->op = FUSE_READ;
+            setctx->count = 2;
+            setctx->iovec = iovec_in;
+            setctx->iobuf = iobuf_in;
+
+            io_uring_sqe_set_data(sqe, setctx);
+
+        resubmit:
+            ret = io_uring_submit(&priv->ring);
+            if (ret < 0) {
+                gf_log("this->name", GF_LOG_ERROR,
+                       "Failed to submit the request, ret = %d", ret);
+                sleep(10);
+                goto resubmit;
+            }
+        }
+        pthread_mutex_unlock(&priv->sq_mutex);
+
+    wait_cqe:
+        ret = io_uring_wait_cqe(&priv->ring, &cqe);
+        if (ret < 0) {
+            if (ret == -EINTR || ret == -EINVAL) {
+                gf_log("this->name", GF_LOG_WARNING,
```

Comment:
The log could be flooded by  this message.

Reply:
will make it GF_LOG_DEBUG.